README.rst

```diff
@@ -1,92 +1,126 @@
+(look for the settings starting with ``email.``).
+
+By default, Sydent will listen on ``0.0.0.0:8090``. This can be changed by changing the values for
+the configuration settings ``clientapi.http.bind_address`` and ``clientapi.http.port``.
 
-The matrix_is_test directory contains sydent's launcher for matrix_is_tester: this needs to be on the
-python path.
+Sydent uses SQLite as its database backend. By default, it will create the database as ``sydent.db``
+in its working directory. You can override that by setting the ``db.file`` configuration setting.
+Sydent is known to be working with SQLite version 3.16.2 and later.
 
-Requests
-========
+SMS originators
+---------------
 
-The requests that synapse servers and clients submit to the identity server are, briefly, as follows:
+Defaults for SMS originators will not be added to the generated config file, these should
+be added to the ``[sms]`` section of that config file in the form::
 
-Request the validation of your email address::
+    originators.<country code> = <long|short|alpha>:<originator>
+
+Where country code is the numeric country code, or ``default`` to specify the originator
+used for countries not listed. For example, to use a selection of long codes for the
+US/Canada, a short code for the UK and an alphanumertic originator for everywhere else::
 
-    curl -XPOST 'http://localhost:8090/_matrix/identity/api/v1/validate/email/requestToken' -H "Content-Type: application/json" -d '{"email": "matthew@arasphere.net", "client_secret": "abcd", "send_attempt": 1}'
-    {"success": true, "sid": "1"}
+    originators.1 = long:12125552368,long:12125552369
+    originators.44 = short:12345
+    originators.default = alpha:Matrix
 
-(Receive 943258 by mail)
+Testing
+=======
 
-Use this code to validate your email address::
+Sydent uses matrix-is-tester (https://github.com/matrix-org/matrix-is-tester/) to provide
+black-box testing of compliance with the `Matrix Identity Service API <https://matrix.org/docs/spec/identity_service/latest>`_.
+This can be run as follows::
 
-    curl -XPOST 'http://localhost:8090/_matrix/identity/api/v1/validate/email/submitToken' -H "Content-Type: application/json" -d '{"token": "943258", "sid": "1", "client_secret": "abcd"}'
-    {"success": true}
+    pip install git+https://github.com/matrix-org/matrix-is-tester.git
+    trial matrix_is_tester
 
-Use the validated email address to bind it to a matrix ID::
+The ``SYDENT_PYTHON`` enviroment variable can be set to launch Sydent with a specific python binary::
 
-    curl -XPOST 'http://localhost:8090/_matrix/identity/api/v1/3pid/bind' -H "Content-Type: application/json" -d '{"sid": "1", "client_secret": "abcd", "mxid": "%40matthew%3amatrix.org"}'
+    SYDENT_PYTHON=/path/to/python trial matrix_is_tester
 
-Lookup::
+The ``matrix_is_test`` directory contains Sydent's launcher for ``matrix_is_tester``: this means
+that Sydent's directory needs to be on the Python path (e.g. ``PYTHONPATH=/path/to/sydent``).
```

Comment:
`PYTHONPATH=$PYTHONPATH:/path/to/sydent` perhaps to not override existing path vars?

Reply:
Not sure, `PYTHONPATH=/home/brendan/Documents/matrix/sydent` works on my box.

Reply:
Right, but it would override whatever you currently have in your `PYTHONPATH`, whereas appending it to the current `PYTHONPATH` seems safer.

Reply:
Yeah, fair enough.

Reply:
Fixed in 0d9c6a0